File: examples/swagger-2/src/test/resources/asserts/typescript/typescript_feign_func/src/evt/CreateOrderEvt.ts
/* tslint:disable */
/* eslint-disable */

        import {BaseEvt} from "./BaseEvt";

    /**
        * 创建订单
    **/


export interface  CreateOrderEvt extends BaseEvt {

            /**
                *属性说明：订单ns，示例输入：test method
                *sn 约束条件：输入字符串的最小长度为：0，输入字符串的最大长度为：50
                *字段在java中的类型为：String
            **/
        sn: string;
            /**
                *属性说明：订单总价，示例输入：
                *totalAmount 约束条件：为必填项，不能为空
                *字段在java中的类型为：Integer
            **/
        totalAmount: number;
}

File: examples/swagger-3-maven-plugin/src/main/java/com/wuxp/codegen/swagger3/example/maven/evt/CreateOrderEvt.java
package com.wuxp.codegen.swagger3.example.maven.evt;


import com.wuxp.codegen.swagger3.example.maven.enums.Sex;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.beans.Transient;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Map;

@Data
public class CreateOrderEvt extends BaseEvt {

    private static DateFormat dateFormat = new SimpleDateFormat();

    @Size(max = 50)
    private String sn;

    @NotNull
    private transient Integer totalAmount;

    private Map<Sex, String> test;

    @Transient
    public Integer getTotalAmount() {
        return totalAmount;
    }
}
